packages/stylelint-config/rules/extras.js

```diff
@@ -0,0 +1,87 @@
+module.exports = {
+  rules: {
+    // Reason: Avoid compound selectors creation using nested ampersand SASS syntax.
+    // Main reasons why use this pattern:
+    // - Unification with design system components code style
+    // - Look up easily for the selector using fulltext search in IDEs
+    // - Better orientation in longer component's code when scrolled down
+    // - Makes `no-descending-specificity` stylelint rule work correctly
+    // - IDE's support to click through from selector usage to its source
+    // Docs: https://stylelint.io/user-guide/rules/list/selector-nested-pattern/
+    'selector-nested-pattern': [
+      '(^&:)|(^&\\.)|(^&\\[)',
```

Comment:
If I'm not mistaken, this can be more explicit:

```suggestion
      '(^&:)|(^&\\.(is\-|has\-))|(^&\\[)',
```

If something like `&.disabled` is there in the wild, I believe it can be changed to `&.is-disabled` (I was thinking [Spirit Button](https://github.com/lmc-eu/spirit-design-system/tree/main/packages/web/src/scss/components/Button) was the reason but it doesn't seem so.).

Reply:
Yes this should be very handy. Regular expression modified.